File: kafis-practices-frontend/libs/interfaces/src/lib/teacher.ts
import { Student } from './student';
import { PracticeDates } from './practice-dates';

export interface Teacher {
  id: number;
  fullName: string;
  position: string;
  userId: number;
  practiceDatesId: number;
  practiceDates: PracticeDates;
  isDeleted: boolean;
  students: Student[];
}

File: kafis-practices-backend/Domain.Core/Entities/Teacher.cs
using Practice.Domain.Core.Entities.Base;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Practice.Domain.Core.Entities
{
    public class Teacher : BaseEntity
    {
        public Guid UserId { get; set; }
        [Required]
        [Column(TypeName = "nvarchar(100)")]
        public string FullName { get; set; }
        [Required]
        [Column(TypeName = "nvarchar(50)")]
        public string Position { get; set; }

        public User User { get; set; }
        public IEnumerable<Student> Students { get; set; }
    }
}
